lib/datadog.ts

```diff
@@ -8,28 +8,45 @@
 
 import * as cdk from "@aws-cdk/core";
 import * as lambda from "@aws-cdk/aws-lambda";
-import { applyLayers, redirectHandlers, addForwarder } from "./index";
+import { applyLayers, redirectHandlers, addForwarder, applyEnvVariables, defaultEnvVar } from "./index";
+import { Transport } from "./transport";
 
 export interface DatadogProps {
   pythonLayerVersion?: number;
   nodeLayerVersion?: number;
   addLayers?: boolean;
   forwarderARN?: string;
+  flushMetricsToLogs?: boolean;
+  site?: string;
+  apiKey?: string;
+  apiKMSKey?: string;
+  enableDDTracing?: boolean;
+  injectLogContext?: boolean;
 }
 
 export class Datadog extends cdk.Construct {
-  /** allows accessing the counter function */
   scope: cdk.Construct;
   props: DatadogProps;
+  xport: Transport;
   constructor(scope: cdk.Construct, id: string, props: DatadogProps) {
     super(scope, id);
     this.scope = scope;
     this.props = props;
+    this.xport = new Transport(this.props.flushMetricsToLogs, this.props.site, this.props.apiKey, this.props.apiKMSKey)
   }
 
-  public addLambdaFunctions(lambdaFunctions: lambda.Function[]) {
+  public addLambdaFunctions(lambdaFunctions: lambda.Function[]){
     if (this.props.addLayers === undefined) {
-      this.props.addLayers = true;
+      this.props.addLayers = defaultEnvVar.addLayers;
+    }
+    if (this.props.enableDDTracing === undefined) {
+      this.props.enableDDTracing = defaultEnvVar.enableDDTracing;
+    }
+    if ((this.props.enableDDTracing === true || this.props.enableDDTracing === undefined) && this.props.forwarderARN === undefined) {
+      throw new Error('A forwarderARN of the Datadog forwarder lambda function is required for Datadog Tracing (enabled by default). This can be disabled by setting enableDDTracing: false.')
```

Comment:
There are multiple ways to set the forwarderARN, not just through this plugin. We should keep it optional here and not throw an exception.

Reply:
Gotcha, removed this logic from the file.